Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -3536,7 +3536,7 @@ final class BuildPlanTests: XCTestCase {
             toolset: .init(
                 knownTools: [
                     .cCompiler: .init(extraCLIOptions: ["-I/fake/sdk/sysroot", "-clang-flag-from-json"]),
-                    .swiftCompiler: .init(extraCLIOptions: ["-swift-flag-from-json"])
+                    .swiftCompiler: .init(extraCLIOptions: ["-swift-flag-from-json", "-use-ld=lld"]),
```

Comment:
is this a windows only change, or impacting on all platforms?

Reply:
It will be used on all the platforms.  The issue is that we were previously relying on leaking `ld` on the host and relying on that existing on Linux and macOS.

Reply:
but `lld` does not exists everywhere either?

Reply:
Ah, using `-use-ld=lld` will change the linker that is picked up and use `lld` which is part of the toolchain.  Let me see if I can instead add `.linker: .init(path: "/fake/toolchain/not-a-linker")`